### ISSUE TITLE
validation result don't match the repo's show

### ISSUE THREAD

**Issue by wenhui-ml** [ultralytics/ultralytics#10379]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I use the yolov8l pretrained model to run CLI just as below, but get de final result don't match the yolo8l's offical mAP50-95 52.9, only 0.519.
Also, I use the yolov9c pretrained model to run CLI just as below, but get de final result don't match the yolo9c's mAP50-95 53.0, only 0.519.
"""
 yolo detect val model=yolov8l.pt imgsz=640 data=coco_val.yaml conf=0.001 iou=0.6 batch=32
Ultralytics YOLOv8.2.2 🚀 Python-3.10.12 torch-2.2.2+cu121 CUDA:0 (NVIDIA GeForce RTX 3090, 24253MiB)
YOLOv8l summary (fused): 268 layers, 43668288 parameters, 0 gradients, 165.2 GFLOPs
val: Scanning /mnt/workspace/cwh8szh/Documents/coco2017/val2017.cache... 5000 images, 48 backgrounds, 0 corrupt: 100%|██████████| 5000/5000 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 157/157 [00:53<00:00,  2.95it/s]
                   all       5000      36781      0.733      0.639      0.695      0.519
                person       5000      11004      0.821      0.756      0.833      0.618
               bicycle       5000        316       0.79      0.596      0.691      0.419
                   car       5000       1932      0.767      0.671       0.74      0.508
            motorcycle       5000        371      0.791      0.728      0.798      0.547
              airplane       5000        143      0.879      0.895      0.938      0.783
                   bus       5000        285      0.826      0.807      0.875      0.766
                 train       5000        190      0.879      0.921      0.945      0.783
                 truck       5000        415       0.65      0.555      0.641      0.474
                  boat       5000        430       0.71      0.507      0.595      0.352
         traffic light       5000        637      0.729      0.499      0.579       0.31
          fire hydrant       5000        101      0.932      0.851        0.9      0.753
             stop sign       5000         75       0.79       0.76      0.802      0.713
         parking meter       5000         60      0.804      0.684      0.716      0.548
                 bench       5000        413      0.646      0.453      0.485      0.357
                  bird       5000        440      0.775      0.527      0.602      0.415
                   cat       5000        202      0.913      0.937      0.956      0.827
                   dog       5000        218      0.863      0.839      0.872      0.762
                 horse       5000        273      0.846      0.857      0.914       0.72
                 sheep       5000        361      0.754      0.788      0.827      0.624
                   cow       5000        380      0.834      0.804      0.866      0.664
              elephant       5000        255      0.837      0.907      0.904      0.738
                  bear       5000         71      0.905      0.934      0.959      0.816
                 zebra       5000        268      0.879      0.877      0.943      0.782
               giraffe       5000        232       0.93      0.905      0.941      0.782
              backpack       5000        371      0.569       0.35      0.403      0.231
              umbrella       5000        413      0.702      0.683      0.734      0.523
               handbag       5000        540      0.578      0.348      0.413      0.246
                   tie       5000        254      0.754      0.551      0.629       0.44
              suitcase       5000        303      0.711       0.68      0.738      0.529
               frisbee       5000        115      0.896       0.87      0.911      0.717
                  skis       5000        241      0.708      0.539      0.604      0.369
             snowboard       5000         69      0.691      0.551      0.649      0.482
           sports ball       5000        263      0.757      0.605      0.663      0.453
                  kite       5000        336      0.699      0.634      0.677      0.462
          baseball bat       5000        146      0.796      0.664      0.739      0.485
        baseball glove       5000        148      0.801      0.624      0.707      0.444
            skateboard       5000        179      0.874      0.849      0.868      0.655
             surfboard       5000        269      0.828      0.632      0.729       0.49
         tennis racket       5000        225        0.9       0.88      0.911      0.643
                bottle       5000       1025      0.683      0.562      0.638      0.444
            wine glass       5000        343      0.781      0.582      0.679      0.465
                   cup       5000        899      0.678      0.618      0.678      0.515
                  fork       5000        215      0.686      0.623      0.692      0.519
                 knife       5000        326      0.695       0.42      0.509      0.322
                 spoon       5000        253      0.593      0.443       0.46      0.329
                  bowl       5000        626      0.645      0.623      0.654      0.504
                banana       5000        379      0.591      0.381      0.474      0.309
                 apple       5000        239      0.528      0.339      0.349      0.248
              sandwich       5000        177      0.651      0.633      0.633      0.485
                orange       5000        287      0.536      0.366       0.44      0.331
              broccoli       5000        316      0.551       0.38       0.45      0.271
                carrot       5000        371      0.459      0.348        0.4      0.259
               hot dog       5000        127      0.711      0.465      0.575       0.44
                 pizza       5000        285      0.764      0.737      0.796      0.627
                 donut       5000        338      0.758      0.595      0.697      0.556
                  cake       5000        316      0.675      0.604      0.677      0.466
                 chair       5000       1791      0.683      0.518      0.604      0.413
                 couch       5000        261      0.649      0.637       0.69      0.551
          potted plant       5000        343      0.578      0.536      0.548      0.349
                   bed       5000        163       0.73      0.693       0.76      0.571
          dining table       5000        697       0.58      0.511      0.528       0.38
                toilet       5000        179      0.799      0.816      0.869      0.714
                    tv       5000        288      0.826      0.776      0.861      0.677
                laptop       5000        231       0.81      0.814       0.87      0.752
                 mouse       5000        106      0.826      0.808      0.859      0.666
                remote       5000        283      0.717      0.583      0.658      0.446
              keyboard       5000        153      0.744      0.684      0.787      0.608
            cell phone       5000        262      0.684       0.63      0.658      0.474
             microwave       5000         55      0.727      0.891      0.885      0.706
                  oven       5000        143      0.663      0.566      0.647      0.472
               toaster       5000          9      0.528      0.667      0.497      0.377
                  sink       5000        225      0.679      0.649      0.686      0.478
          refrigerator       5000        126      0.799       0.79      0.831      0.712
                  book       5000       1161      0.588      0.211      0.331      0.191
                 clock       5000        267      0.815       0.74      0.781      0.543
                  vase       5000        277      0.668      0.596      0.666      0.475
              scissors       5000         36      0.762      0.472      0.536      0.454
            teddy bear       5000        191      0.766      0.675      0.748       0.62
            hair drier       5000         11      0.659      0.182      0.264      0.207
            toothbrush       5000         57      0.592      0.456      0.497      0.379
Speed: 0.1ms preprocess, 6.1ms inference, 0.0ms loss, 0.8ms postprocess per image
Saving runs/detect/val4/predictions.json...
"""
"""
yolo detect val model=yolov9c.pt imgsz=640 data=coco_val.yaml conf=0.001 iou=0.6 batch=32
Ultralytics YOLOv8.2.2 🚀 Python-3.10.12 torch-2.2.2+cu121 CUDA:0 (NVIDIA GeForce RTX 3090, 24253MiB)
YOLOv9c summary (fused): 384 layers, 25380928 parameters, 0 gradients, 102.7 GFLOPs
val: Scanning /mnt/workspace/cwh8szh/Documents/coco2017/val2017.cache... 5000 images, 48 backgrounds, 0 corrupt: 100%|██████████| 5000/5000 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 157/157 [00:47<00:00,  3.33it/s]
                   all       5000      36781      0.731      0.634      0.694      0.519
                person       5000      11004      0.829      0.743       0.83      0.613
               bicycle       5000        316      0.765       0.56      0.666      0.425
                   car       5000       1932      0.765      0.679      0.742      0.505
            motorcycle       5000        371      0.798      0.712      0.793      0.541
              airplane       5000        143      0.878      0.895      0.934      0.796
                   bus       5000        285      0.844        0.8      0.876       0.76
                 train       5000        190      0.885      0.887      0.942      0.777
                 truck       5000        415      0.699      0.557      0.655       0.48
                  boat       5000        430      0.725      0.497      0.577      0.347
         traffic light       5000        637      0.724      0.493      0.577      0.312
          fire hydrant       5000        101      0.956      0.861      0.925      0.771
             stop sign       5000         75      0.816       0.72      0.787      0.696
         parking meter       5000         60      0.783       0.65       0.69      0.535
                 bench       5000        413      0.637      0.416      0.478      0.345
                  bird       5000        440      0.773      0.477      0.595      0.409
                   cat       5000        202      0.925      0.922      0.958       0.82
                   dog       5000        218      0.846      0.839      0.868      0.747
                 horse       5000        273      0.874      0.836      0.903      0.718
                 sheep       5000        361      0.752      0.781      0.832      0.621
                   cow       5000        380      0.844      0.769       0.85      0.662
              elephant       5000        255      0.827       0.91      0.892      0.731
                  bear       5000         71      0.926       0.93      0.953      0.826
                 zebra       5000        268      0.905      0.884      0.937      0.764
               giraffe       5000        232      0.923      0.924      0.936      0.793
              backpack       5000        371      0.575      0.348      0.396       0.23
              umbrella       5000        413      0.711      0.673       0.73      0.516
               handbag       5000        540      0.579      0.339      0.402      0.239
                   tie       5000        254      0.808      0.531      0.621      0.424
              suitcase       5000        303      0.688       0.66      0.722      0.523
               frisbee       5000        115      0.887      0.904      0.921      0.725
                  skis       5000        241      0.733      0.527        0.6      0.349
             snowboard       5000         69      0.719       0.58      0.639      0.481
           sports ball       5000        263       0.82      0.578      0.651      0.446
                  kite       5000        336       0.67      0.587      0.666      0.465
          baseball bat       5000        146      0.823      0.668      0.748      0.488
        baseball glove       5000        148      0.809      0.649      0.707      0.449
            skateboard       5000        179      0.872      0.838      0.842      0.638
             surfboard       5000        269       0.84      0.626      0.725      0.495
         tennis racket       5000        225      0.876       0.85      0.901      0.634
                bottle       5000       1025      0.693      0.584      0.643      0.448
            wine glass       5000        343      0.803      0.586      0.678      0.451
                   cup       5000        899      0.678      0.618      0.687      0.517
                  fork       5000        215      0.709      0.619        0.7      0.529
                 knife       5000        326      0.619      0.411      0.471      0.313
                 spoon       5000        253       0.62      0.458      0.484      0.331
                  bowl       5000        626      0.657      0.618       0.66      0.511
                banana       5000        379      0.591      0.356      0.439       0.29
                 apple       5000        239      0.531      0.322       0.36      0.251
              sandwich       5000        177      0.601      0.612      0.626      0.499
                orange       5000        287      0.549      0.376       0.44      0.341
              broccoli       5000        316      0.604      0.415      0.451      0.267
                carrot       5000        371      0.523      0.372      0.398      0.266
               hot dog       5000        127      0.725      0.512      0.631      0.486
                 pizza       5000        285      0.766       0.74      0.802       0.63
                 donut       5000        338      0.699      0.595      0.688      0.545
                  cake       5000        316      0.646      0.595      0.649      0.439
                 chair       5000       1791      0.679      0.514      0.604      0.415
                 couch       5000        261      0.674      0.626      0.704      0.552
          potted plant       5000        343      0.585      0.519      0.552      0.344
                   bed       5000        163      0.714      0.675      0.752      0.573
          dining table       5000        697      0.579      0.515      0.538      0.376
                toilet       5000        179      0.797      0.838      0.865      0.719
                    tv       5000        288      0.812      0.785      0.855      0.683
                laptop       5000        231      0.801      0.788      0.858       0.75
                 mouse       5000        106      0.809      0.801      0.849      0.664
                remote       5000        283      0.699      0.591      0.664      0.434
              keyboard       5000        153      0.723      0.719      0.801      0.632
            cell phone       5000        262      0.656      0.612      0.639      0.456
             microwave       5000         55      0.782        0.8      0.846      0.706
                  oven       5000        143      0.622      0.594      0.631      0.451
               toaster       5000          9      0.481      0.778      0.786      0.573
                  sink       5000        225      0.672      0.622       0.67      0.457
          refrigerator       5000        126       0.77      0.722      0.794      0.685
                  book       5000       1161      0.604      0.208      0.319      0.181
                 clock       5000        267      0.797      0.736      0.775      0.556
                  vase       5000        277      0.663      0.592      0.626      0.448
              scissors       5000         36      0.799      0.444      0.576      0.463
            teddy bear       5000        191      0.722      0.723      0.764      0.618
            hair drier       5000         11      0.253     0.0909      0.195      0.149
            toothbrush       5000         57      0.644      0.572      0.546      0.401
Speed: 0.1ms preprocess, 5.2ms inference, 0.0ms loss, 0.7ms postprocess per image
Saving runs/detect/val5/predictions.json...
"""

**Comment by glenn-jocher** [comment]
@wenhui-ml hey there! Thanks for reaching out and running those validations. 🙌 It looks like the discrepancy you're seeing between your validation results and the official mAP scores might be due to slight differences in the validation configuration or dataset version.

The mAP score difference you're observing, e.g., `0.519` vs the official `0.529` for YOLOv8l, is very minor. Such variations can occur due to factors like how the dataset is split or the specific hardware and software environment.

Regarding your CLI command, it appears correct. However, ensure you're using the same validation dataset (`coco_val.yaml`) as the one used for reporting the official metrics. Dataset versions or annotations could have minor differences impacting the mAP.

If the slight variation persists and you're looking to exactly match the reported mAP, consider checking:

- The exact dataset and splits used for the official results. Ensure no updates or discrepancies in the dataset version.
- The hardware and software environment, as this can also introduce minor variations.

For most practical purposes, a small deviation in mAP like this shouldn't significantly impact the deployment or performance of your models. If your use case requires pinpoint accuracy in validation metrics, I'd recommend directly reaching out or checking if there have been follow-up discussions or notes regarding the reported metrics.

Feel free to keep us posted on your progress or if there's anything more we can help with. Keep up the great work! 🚀

**Comment by wenhui-ml** [comment]
many thanks for your nice answer

**Comment by glenn-jocher** [comment]
@wenhui-ml you're welcome! 😊 If you have any more questions or need further assistance, feel free to ask. Happy coding! 🚀
